### ISSUE TITLE
error

### ISSUE THREAD

**Issue by faisalkhan323** [ThrowTheSwitch/Ceedling#496]
I downloaded built in code from https://github.com/ElectronVector/try-tdd-with-ceedling
but when i run ceedling test:all OR rake test:all  command I got an error. 
[error.docx](https://github.com/ThrowTheSwitch/Ceedling/files/4752908/error.docx)

Please let me know about problem. Thanks in advance
ruby 2.6.6p146 (2020-03-31 revision 67876) [x64-mingw32]

C:\Users\Junaid Computers>cd try-tdd-with-ceedling-master

C:\Users\Junaid Computers\try-tdd-with-ceedling-master>ceedling test:all

Test 'test_lights.c'
--------------------
gcc.exe: error: Computers/try-tdd-with-ceedling-master/vendor/ceedling/vendor/unity/src: No such file or directory
gcc.exe: error: Computers/try-tdd-with-ceedling-master/vendor/ceedling/vendor/cmock/src: No such file or directory
ERROR: Shell command failed.
> Shell executed command:
'gcc.exe -E -MM -MG -Itest -Itest/support -Isrc -IC:/Users/Junaid Computers/try-tdd-with-ceedling-master/vendor/ceedling/vendor/unity/src -IC:/Users/Junaid Computers/try-tdd-with-ceedling-master/vendor/ceedling/vendor/cmock/src -Ibuild/test/mocks -DTEST -DTEST -DGNU_COMPILER -w "build/temp/_test_lights.c"'
> And exited with status: [1].

rake aborted!
ShellExecutionException: ShellExecutionException
C:/Users/Junaid Computers/try-tdd-with-ceedling-master/vendor/ceedling/lib/ceedling/tool_executor.rb:74:in `exec'
C:/Users/Junaid Computers/try-tdd-with-ceedling-master/vendor/ceedling/lib/ceedling/preprocessinator_includes_handler.rb:39:in `form_shallow_dependencies_rule'
C:/Users/Junaid Computers/try-tdd-with-ceedling-master/vendor/ceedling/lib/ceedling/preprocessinator.rb:31:in `preprocess_shallow_includes'
C:/Users/Junaid Computers/try-tdd-with-ceedling-master/vendor/ceedling/lib/ceedling/preprocessinator.rb:11:in `block in setup'
C:/Users/Junaid Computers/try-tdd-with-ceedling-master/vendor/ceedling/lib/ceedling/preprocessinator_helper.rb:11:in `preprocess_includes'
C:/Users/Junaid Computers/try-tdd-with-ceedling-master/vendor/ceedling/lib/ceedling/preprocessinator.rb:17:in `preprocess_test_and_invoke_test_mocks'
C:/Users/Junaid Computers/try-tdd-with-ceedling-master/vendor/ceedling/lib/ceedling/test_invoker.rb:42:in `block in setup_and_invoke'
C:/Users/Junaid Computers/try-tdd-with-ceedling-master/vendor/ceedling/lib/ceedling/test_invoker.rb:32:in `setup_and_invoke'
C:/Users/Junaid Computers/try-tdd-with-ceedling-master/vendor/ceedling/lib/ceedling/tasks_tests.rake:8:in `block (2 levels) in <top (required)>'
C:/Ruby26-x64/bin/ceedling:23:in `load'
C:/Ruby26-x64/bin/ceedling:23:in `<main>'
Tasks: TOP => test:all
(See full trace by running task with --trace)

--------------------
OVERALL TEST SUMMARY
--------------------

No tests executed.

C:\Users\Junaid Computers\try-tdd-with-ceedling-master>

**Comment by mvandervoord** [comment]
I suspect that when you created the project, you went with the defaults, which are to use the gem's version of cmock and unity. The project.yml file you're using seems to be expecting that they be in the vendor directory. You can accomplish this by passing `--local` when you create the project. 

If you'd like to adjust the current project file to be correct, adjust these two fields( the first is something to add to the `:project` section at the top. The latter is a field to modify in the `:plugins` section at the bottom):

```
:project:
  :which_ceedling: gem
you can go with the default and remove the following line from your project.yml:

:plugins:
  :load_paths:
    - "#{Ceedling.load_path}"
```

**Comment by faisalkhan323** [comment]
thank you for you reply.but I am facing the same error .



details of my project.yml file are as follows---# Notes:
# Sample project C code is not presently written to produce a release artifact.
# As such, release build options are disabled.
# This sample, therefore, only demonstrates running a collection of unit tests.:project:
  :which_ceedling: gem
  :use_exceptions: FALSE
  :use_test_preprocessor: TRUE
  :use_auxiliary_dependencies: TRUE
  :build_root: build
#  :release_build: TRUE
  :test_file_prefix: test_#:release_build:
#  :output: MyApp.out
#  :use_assembly: FALSE:environment::extension:
  :executable: .out:paths:
  :test:
    - +:test/**
    - -:test/support
  :source:
    - src/**
  :support:
    - test/support:defines:
  # in order to add common defines:
  #  1) remove the trailing [] from the :common: section
  #  2) add entries to the :common: section (e.g. :test: has TEST defined)
  :commmon: &common_defines []
  :test:
    - *common_defines
    - TEST
  :test_preprocess:
    - *common_defines
    - TEST:cmock:
  :mock_prefix: mock_
  :when_no_prototypes: :warn
  :enforce_strict_ordering: TRUE
  :plugins:
    - :ignore
    - :callback
  :treat_as:
    uint8:    HEX8
    uint16:   HEX16
    uint32:   UINT32
    int8:     INT8
    bool:     UINT8#:tools:
# Ceedling defaults to using gcc for compiling, linking, etc.
# As [:tools] is blank, gcc will be used (so long as it's in your system path)
# See documentation to configure a given toolchain for use:plugins:
  :load_paths:
    - vendor/ceedling/plugins
  :enabled:
    - stdout_pretty_tests_report
    - module_generator
...

    On Tuesday, June 9, 2020, 08:29:07 PM GMT+5, Mark VanderVoord <notifications@github.com> wrote:  
 
 


I suspect that when you created the project, you went with the defaults, which are to use the gem's version of cmock and unity. The project.yml file you're using seems to be expecting that they be in the vendor directory. You can accomplish this by passing --local when you create the project.

If you'd like to adjust the current project file to be correct, adjust these two fields( the first is something to add to the :project section at the top. The latter is a field to modify in the :plugins section at the bottom):
:project:
  :which_ceedling: gem
you can go with the default and remove the following line from your project.yml:

:plugins:
  :load_paths:
    - "#{Ceedling.load_path}"

**Comment by mvandervoord** [comment]
The `rake test:all` version no longer exists, so you can ignore that... in fact, remove `rakefile.rb` for good measure.

You've not update the :plugins: section. You still have this:

```
:plugins:
  :load_paths:
    - vendor/ceedling/plugins
```

but you want this:

```
:plugins:
  :load_paths:
    - "#{Ceedling.load_path}"
```

If that doesn't fix things for you, there are two helpful things to do in your response. (1) You can tell me the version you're working with (ask by using `ceedling version`). (2) If you copy and paste code or command line responses, it's helpful to add a line BEFORE the snippet with three backticks and a line AFTER the snippet with three backticks. This will keep github's markdown engine from mangling your text.

**Comment by faisalkhan323** [comment]
thank you for guidance but still there is error.



details of project file is...
---# Notes:
# Sample project C code is not presently written to produce a release artifact.
# As such, release build options are disabled.
# This sample, therefore, only demonstrates running a collection of unit tests.:project:
  :which_ceedling: gem
  :use_exceptions: FALSE
  :use_test_preprocessor: TRUE
  :use_auxiliary_dependencies: TRUE
  :build_root: build
#  :release_build: TRUE
  :test_file_prefix: test_#:release_build:
#  :output: MyApp.out
#  :use_assembly: FALSE:environment::extension:
  :executable: .out:paths:
  :test:
    - +:test/**
    - -:test/support
  :source:
    - src/**
  :support:
    - test/support:defines:
  # in order to add common defines:
  #  1) remove the trailing [] from the :common: section
  #  2) add entries to the :common: section (e.g. :test: has TEST defined)
  :commmon: &common_defines []
  :test:
    - *common_defines
    - TEST
  :test_preprocess:
    - *common_defines
    - TEST:cmock:
  :mock_prefix: mock_
  :when_no_prototypes: :warn
  :enforce_strict_ordering: TRUE
  :plugins:
    - :ignore
    - :callback
  :treat_as:
    uint8:    HEX8
    uint16:   HEX16
    uint32:   UINT32
    int8:     INT8
    bool:     UINT8#:tools:
# Ceedling defaults to using gcc for compiling, linking, etc.
# As [:tools] is blank, gcc will be used (so long as it's in your system path)
# See documentation to configure a given toolchain for use:plugins:
  :load_paths:
  - Ceedling.load_path
  :enabled:
    - stdout_pretty_tests_report
    - module_generator
...



    On Tuesday, June 9, 2020, 09:42:01 PM GMT+5, Mark VanderVoord <notifications@github.com> wrote:  
 
 


The rake test:all version no longer exists, so you can ignore that... in fact, remove rakefile.rb for good measure.

You've not update the :plugins: section. You still have this:
:plugins:
  :load_paths:
    - vendor/ceedling/plugins

but you want this:
:plugins:
  :load_paths:
    - "#{Ceedling.load_path}"

If that doesn't fix things for you, there are two helpful things to do in your response. (1) You can tell me the version you're working with (ask by using ceedling version). (2) If you copy and paste code or command line responses, it's helpful to add a line BEFORE the snippet with three backticks and a line AFTER the snippet with three backticks. This will keep github's markdown engine from mangling your text.

**Comment by mvandervoord** [comment]
:)

Yes, `Ceedling.load_path` is not the same as ` "#{Ceedling.load_path}" `

I suspect that's the issue.

**Comment by faisalkhan323** [comment]
same error
---# Notes:
# Sample project C code is not presently written to produce a release artifact.
# As such, release build options are disabled.
# This sample, therefore, only demonstrates running a collection of unit tests.:project:
  :which_ceedling: gem
  :use_exceptions: FALSE
  :use_test_preprocessor: TRUE
  :use_auxiliary_dependencies: TRUE
  :build_root: build
#  :release_build: TRUE
  :test_file_prefix: test_#:release_build:
#  :output: MyApp.out
#  :use_assembly: FALSE:environment::extension:
  :executable: .out:paths:
  :test:
    - +:test/**
    - -:test/support
  :source:
    - src/**
  :support:
    - test/support:defines:
  # in order to add common defines:
  #  1) remove the trailing [] from the :common: section
  #  2) add entries to the :common: section (e.g. :test: has TEST defined)
  :commmon: &common_defines []
  :test:
    - *common_defines
    - TEST
  :test_preprocess:
    - *common_defines
    - TEST:cmock:
  :mock_prefix: mock_
  :when_no_prototypes: :warn
  :enforce_strict_ordering: TRUE
  :plugins:
    - :ignore
    - :callback
  :treat_as:
    uint8:    HEX8
    uint16:   HEX16
    uint32:   UINT32
    int8:     INT8
    bool:     UINT8#:tools:
# Ceedling defaults to using gcc for compiling, linking, etc.
# As [:tools] is blank, gcc will be used (so long as it's in your system path)
# See documentation to configure a given toolchain for use:plugins:
  :load_paths:
  -  "#{Ceedling.load_path}"
  :enabled:
    - stdout_pretty_tests_report
    - module_generator
...


    On Tuesday, June 9, 2020, 10:37:27 PM GMT+5, Mark VanderVoord <notifications@github.com> wrote:  
 
 


:)

Yes, Ceedling.load_path is not the same as "#{Ceedling.load_path}"

I suspect that's the issue.

**Comment by mvandervoord** [comment]
1. What do you get when you issue `ceedling version`?
2. Have you removed `rakefile.rb` yet?
3. What is the error you receive when you issue `ceedling clobber test:all` at this point?

**Comment by faisalkhan323** [comment]
it works thanks.but I did not understand what was the problem and how it is solved now.. would you please explain?





    On Tuesday, June 9, 2020, 10:48:38 PM GMT+5, Mark VanderVoord <notifications@github.com> wrote:  
 
 

   
   - What do you get when you issue ceedling version?
   - Have you removed rakefile.rb yet?
   - What is the error you receive when you issue ceedling clobber test:all at this point?

**Comment by faisalkhan323** [comment]
thanks a lot for your help.I have understood and the problem has been solved by doing following steps.

Try tdd ceedlig error 

If we  like to adjust the current project file to becorrect, we have to adjust these two fields( the first is something to add tothe :project section at the top. The latter is a field to modify in the :plugins section at thebottom):

:project:

  :which_ceedling: gem

you can go with the default and remove the following line fromyour project.yml:

 

:plugins:

  :load_paths:

    -"#{Ceedling.load_path}"

 

Romove rakefile.rb file.

 
thanks

    On Tuesday, June 9, 2020, 10:57:58 PM GMT+5, Muhammad Faisal Khan <muhammadfaisalkhan323@yahoo.com> wrote:  
 
  
it works thanks.but I did not understand what was the problem and how it is solved now.. would you please explain?





    On Tuesday, June 9, 2020, 10:48:38 PM GMT+5, Mark VanderVoord <notifications@github.com> wrote:  
 
 

   
   - What do you get when you issue ceedling version?
   - Have you removed rakefile.rb yet?
   - What is the error you receive when you issue ceedling clobber test:all at this point?

**Comment by mvandervoord** [comment]
Glad to hear you have it working. 

Those three changes update the project.yml file in that 5-year-old example to match the latest defaults and paradigms of ceedling. While we try not to break backward compatibility often, we're also on a mission to make ceedling easier to use for newcomers. That has meant some breaking changes occasionally.

**Comment by faisalkhan323** [comment]
hi, Mark Again I am facing some problems related to ceedling. I have posted my problem in my  get hub account.mine issue number is 515. I would be very thankful to you if I have your kind considerations on that issue.kind regardsfaisal

    On ‎Wednesday‎, ‎June‎ ‎10‎, ‎2020‎ ‎02‎:‎35‎:‎28‎ ‎AM‎ ‎CST, Mark VanderVoord <notifications@github.com> wrote:  
 
 


Closed #496.
